File: src/main/java/com/innercircle/survey/domain/Option.java
package com.innercircle.survey.domain;

import jakarta.persistence.*;
import lombok.*;

@Getter
@Entity
@Builder
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
public class Option {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "option_id")
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "question_id")
    private Question question;

    private String text;

    public void update(String text) {
        this.text = text;
    }
}

File: src/main/kotlin/org/example/survey/domain/Option.kt
package org.example.survey.domain

import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table

@Entity
@Table(name = "option")
class Option(
    question: Question,
    content: String,
    seq: Int,
) {
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "question_id")
    var question: Question = question

    val content: String = content
    val seq: Int = seq

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val optionId: Long = 0
}
